src/ImageSharp/Common/Helpers/DenseMatrixUtils.cs

```diff
@@ -247,6 +248,91 @@ internal static class DenseMatrixUtils
             target = vector;
         }
 
+        /// <summary>
+        /// Computes the sum of vectors in <paramref name="targetRow"/> weighted by the kernel weight values.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="kernel">The 1D convolution kernel.</param>
+        /// <param name="sourcePixels">The source frame.</param>
+        /// <param name="targetRow">The target row.</param>
+        /// <param name="row">The current row.</param>
+        /// <param name="column">The current column.</param>
+        /// <param name="minRow">The minimum working area row.</param>
+        /// <param name="maxRow">The maximum working area row.</param>
+        /// <param name="minColumn">The minimum working area column.</param>
+        /// <param name="maxColumn">The maximum working area column.</param>
+        public static void Convolve1D<TPixel>(
+            Span<Complex64> kernel,
+            Buffer2D<TPixel> sourcePixels,
+            Span<ComplexVector4> targetRow,
+            int row,
+            int column,
+            int minRow,
+            int maxRow,
+            int minColumn,
+            int maxColumn)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            ComplexVector4 vector = default;
+            int kernelLength = kernel.Length;
+            int radiusY = kernelLength >> 1;
+            int sourceOffsetColumnBase = column + minColumn;
+            ref Complex64 baseRef = ref MemoryMarshal.GetReference(kernel);
+
+            for (int i = 0; i < kernelLength; i++)
+            {
+                int offsetY = (row + i - radiusY).Clamp(minRow, maxRow);
+                int offsetX = sourceOffsetColumnBase.Clamp(minColumn, maxColumn);
+                Span<TPixel> sourceRowSpan = sourcePixels.GetRowSpan(offsetY);
+                var currentColor = sourceRowSpan[offsetX].ToVector4();
+
+                vector.Sum(Unsafe.Add(ref baseRef, i) * currentColor);
+            }
+
+            targetRow[column] = vector;
+        }
+
+        /// <summary>
+        /// Computes the sum of vectors in <paramref name="targetRow"/> weighted by the kernel weight values.
+        /// </summary>
+        /// <param name="kernel">The 1D convolution kernel.</param>
+        /// <param name="sourceValues">The source frame.</param>
+        /// <param name="targetRow">The target row.</param>
+        /// <param name="row">The current row.</param>
+        /// <param name="column">The current column.</param>
+        /// <param name="minRow">The minimum working area row.</param>
+        /// <param name="maxRow">The maximum working area row.</param>
+        /// <param name="minColumn">The minimum working area column.</param>
+        /// <param name="maxColumn">The maximum working area column.</param>
+        public static void Convolve1D(
```

Comment:
I'd like to remove these two methods from here as they do not use the `DenseMatrix<T>` struct. Also consider renaming to `Convolve4` to indicate that all four components are convolved.

Reply:
Done in https://github.com/SixLabors/ImageSharp/pull/842/commits/0a77f69ebf043b7d2811360718c280a94b7746f8.